File: src/main/java/test/sombra/type/service/TypeService.java
package test.sombra.type.service;

import org.springframework.http.ResponseEntity;
import test.sombra.type.domain.Type;

import java.util.List;

/**
 * Created by Ivan on 23.03.2017.
 */
public interface TypeService {

    ResponseEntity<List<Type>> getAll();

    ResponseEntity<Type> add(Type type);

    ResponseEntity<Type> update(Type type);

    ResponseEntity<Void> delete(Long id);

    ResponseEntity<Type> getOne(Long id);

    ResponseEntity<Type> getOneByName(String name);
}

File: src/main/webapp/app/type/type.service.js
(function () {
    'use strict';

    angular
        .module('main')
        .service('TypesService', function ($http) {

            var urlBase = '/type';

            this.getAll = function () {
                return $http.get(urlBase);
            };

            this.getOneById = function (id) {
                return $http.get('/type/id/' + id);
            };

            this.getOneByName = function (name) {
                return $http.get(urlBase + '/' + name);
            };

            this.save = function (type) {
                return $http.post(urlBase, type);
            };

            this.update = function (type) {
                return $http.post(urlBase + '/update', type);
            }
        });
})();

